man/coap-client.txt.in

```diff
@@ -118,6 +119,11 @@ OPTIONS
 *-B* seconds::
    Break operation after waiting given seconds (default is 90).
 
+*-C* cafile::
+  PEM file containing the CA Certificate that was used to sign the certfile.
+  This will trigger the validation of the server certificate.  If certfile
+  is self-signed, then also add '-C certfile' on the command line.
```

Comment:
Actually original text was correct in that you need '-c certfile -C certfile' for a self-signed certificate to be able to do certificate verification.  However, text updated for clarity to 
````
*-C* cafile::
  PEM file containing the CA Certificate that was used to sign the certfile
  defined using '-c certfile'.
  This will trigger the validation of the server certificate.
  If certfile is self-signed, then also add '-C certfile' (as this cafile
  'signed' the certfile) on the command line to trigger validation.
````

Reply:
The problem with the last sentence is that it says you need to specific `-C cafile -C certfile`. The code just sets `ca_file = optarg`, i.e. only the last `-C` option will have an effect.

Reply:
How does the following work for you
````
*-C* cafile::
  PEM file containing the CA Certificate that was used to sign the certfile
  defined using *-c certfile*.
  This will trigger the validation of the server certificate.
  If certfile is self-signed (as defined by *-c certfile*), then you need to
  have on the command line the same filename for both the certfile and cafile
  (as in  *-c certfile -C certfile*) to trigger validation.
````

Reply:
Yes, this looks good.